workflow/common/util.go

```diff
@@ -118,8 +118,10 @@ func overwriteWithArguments(argParam, inParam *wfv1.Parameter) {
 	if argParam != nil {
 		if argParam.Value != nil {
 			inParam.Value = argParam.Value
+			inParam.ValueFrom = nil
 		} else {
 			inParam.ValueFrom = argParam.ValueFrom
+			inParam.Value = nil
```

Comment:
This line is not necessary, just add it for clarirty that we do ignore `inParam.Value`